web/client/epics/geostory.js

```diff
@@ -461,6 +474,89 @@ export const handlePendingGeoStoryChanges = action$ =>
                     )
             )
     );
+
+/**
+ * Handle the url updates on currentPage change
+ * @param {Observable} action$ stream of actions
+ * @param {object} store redux store
+ */
+export const urlUpdateOnScroll = (action$, {getState}) =>
+    action$.ofType(UPDATE_CURRENT_PAGE)
+        .debounceTime(50) // little delay if too many UPDATE_CURRENT_PAGE actions come
+        .switchMap(({sectionId, columnId}) => {
+            if (
+                modeSelector(getState()) !== 'edit' && (!!columnId || !!sectionId)
+                && updateUrlOnScrollSelector(getState())
+            ) {
+                const EMPTY = 'EMPTY';
+                const storyId = get(getState(), 'geostory.resource.id');
+                const storyIds = window?.location?.hash?.substring(window?.location?.hash.indexOf(storyId)).split('/');
+                if (sectionId && storyId) {
```

Comment:
To make this logic testable we could externalize this code to a function utils `GeoStoryUtils`.
We could call it `updateGeoStoryPath` that given currentHash, storyId, sectionId and columnId it return the new hash.
At the end we will have something like this in the epic and we can test `updateGeoStoryPath` function in the utils
NOTE: this is an example
```js
const storyId = get(getState(), 'geostory.resource.id');
const hrefWithoutHash = window?.location?.href.split('#')[0];
const currentHash = window?.location?.hash;
const newHash = updateGeoStoryPath(currentHash, { storyId, sectionId, columnId }); // here all the logic to return the new hash
history.pushState(null, '', `${hrefWithoutHash}/${newHash}`);
return Observable.empty();
```